File: src/main/java/com/gitlab/daring/image/features/KeyPointList.java
package com.gitlab.daring.image.features;

import org.bytedeco.javacpp.opencv_core.KeyPointVector;
import org.bytedeco.javacpp.opencv_core.Mat;

public class KeyPointList {

    public final Mat mat;
    public final KeyPointVector points = new KeyPointVector();
    public final Mat descriptors = new Mat();

    public KeyPointList(Mat mat) {
        this.mat = mat;
    }
}

File: src/main/kotlin/com/gitlab/daring/image/features/KeyPointList.kt
package com.gitlab.daring.image.features

import org.bytedeco.javacpp.opencv_core.KeyPointVector
import org.bytedeco.javacpp.opencv_core.Mat

data class KeyPointList(
        val mat: Mat,
        val points: KeyPointVector = KeyPointVector(),
        val descriptors: Mat = Mat()
)